ext/opcache/jit/zend_jit_ir.c

```diff
@@ -3666,34 +3666,22 @@ static int zend_jit_bb_start(zend_jit_ctx *jit, int b)
 		pred = jit->ssa->cfg.predecessors[bb->predecessor_offset];
 		ref = jit->bb_edges[jit->bb_predecessors[b]];
 		if (ref == IR_UNUSED) {
-			if (!jit->ctx.control) {
-				ir_BEGIN(IR_UNUSED); /* unreachable block */
-			}
+			ir_BEGIN(IR_UNUSED); /* unreachable block */
```

Comment:
Hi @nielsdos, thanks for looking into this, but I'm not sure if it costs to do this.
Do yo really understand all the details of your changes or just relay on ``ZEND_ASSERT()`` above?
I'm not completely sure if that assertion can't be triggered at some moment.
Restoration of the removed code at that case may be a much complicated task.

To be sure, I would need to do careful analyses and wouldn't like to do this in this moment.

Can you analyse the possibility of missed 'jit->control`?

Reply:
No, I just relayed the ZEND_ASSERT() from above. I understand why we should not do this. I can try to analyze this in the near future.

Reply:
> I can try to analyze this in the near future.

It would be great if you could add some comments during review and analyses.
Clean-ups are welcome as well. I know the sources are quite dirty...